src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -1509,14 +1504,14 @@ private void CompileSwitchExpression(Expression expr)
                     return;
                 }
 
-                TypeCode switchType = System.Dynamic.Utils.TypeExtensions.GetTypeCode(node.SwitchValue.Type);
+                TypeCode switchType = node.SwitchValue.Type.GetTypeCode();
```

Comment:
Any ETA on getting GetTypeCode back? We had a previous PR to turn quite a few things into extension method syntax; things such as `GetNonNullableType` (our own extension method) still need to be turned into that syntax. At that point, it becomes interesting whether we'd write:

```csharp
type.GetNonNullableType().GetTypeCode()
```
or
```csharp
Type.GetTypeCode(type.GetNonNullableType())
```

The former would be cleaner for sure. I assume a trivial extension method for `GetTypeCode` would get inlined by the JIT, so maybe the extension method isn't too bad?